aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -544,19 +580,48 @@ private void assertNoTerminalStatus(final DBInstance dbInstance) throws CfnNotSt
 
     protected boolean isDBInstanceStabilizedAfterMutate(
             final ProxyClient<RdsClient> rdsProxyClient,
-            final ResourceModel model
+            final ResourceModel model,
+            final CallbackContext context
     ) {
         final DBInstance dbInstance = fetchDBInstance(rdsProxyClient, model);
 
         assertNoTerminalStatus(dbInstance);
 
-        return isDBInstanceAvailable(dbInstance) &&
+        if (isDBInstanceAvailable(dbInstance) &&
                 isReplicationComplete(dbInstance) &&
                 isDBParameterGroupNotApplying(dbInstance) &&
                 isNoPendingChanges(dbInstance) &&
                 isVpcSecurityGroupsActive(dbInstance) &&
                 isDomainMembershipsJoined(dbInstance) &&
-                isPromotionTierUpdated(dbInstance, model);
+                isPromotionTierUpdated(dbInstance, model)
+        ) {
+            assertNoFailedEvents(rdsProxyClient, model, context);
```

Comment:
In theory, we could fail faster by checking events rather then performing the rest of stabilization. It might have more complex interaction with rollbacks, however, since instance might not be ready for modification if we fail fast.